service/src/main/java/uk/gov/hmcts/reform/fpl/service/docmosis/DischargeCareOrderGenerationService.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.hmcts.reform.fpl.service.docmosis;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.config.HmctsCourtLookupConfiguration;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.OrderTypeAndDocument;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.docmosis.DocmosisGeneratedOrder;
+import uk.gov.hmcts.reform.fpl.model.docmosis.DocmosisGeneratedOrder.DocmosisGeneratedOrderBuilder;
+import uk.gov.hmcts.reform.fpl.model.order.generated.GeneratedOrder;
+import uk.gov.hmcts.reform.fpl.service.DischargeCareOrderService;
+
+import java.util.List;
+
+import static java.lang.String.format;
+import static java.lang.System.lineSeparator;
+import static java.util.stream.Collectors.joining;
+import static org.apache.commons.lang3.ObjectUtils.defaultIfNull;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class DischargeCareOrderGenerationService extends GeneratedOrderTemplateDataGeneration {
+
+    private final HmctsCourtLookupConfiguration hmctsCourtLookupConfiguration;
+    private final DischargeCareOrderService dischargeCareOrder;
+
+    @SuppressWarnings("rawtypes")
+    @Override
+    DocmosisGeneratedOrderBuilder populateCustomOrderFields(CaseData caseData) {
+        OrderTypeAndDocument orderTypeAndDocument = caseData.getOrderTypeAndDocument();
+
+        return DocmosisGeneratedOrder.builder()
+            .orderTitle(orderTypeAndDocument.getFullType())
+            .childrenAct("Section 39(1) Children Act 1989")
+            .orderDetails(formatOrderDetails(caseData));
+    }
+
+    @Override
+    List<Element<Child>> getSelectedChildren(CaseData caseData) {
+        return dischargeCareOrder.getSelectedChildren(caseData);
+    }
+
+    private String formatOrderDetails(CaseData caseData) {
+        final List<GeneratedOrder> careOrders = dischargeCareOrder.getSelectedCareOrders(caseData);
+
+        final String defaultCourtName = getDefaultCourt(caseData);
+
+        if (careOrders.size() == 1) {
+            GeneratedOrder careOrder = careOrders.get(0);
+            return format("The court discharges the care order made by the %s on %s",
+                defaultIfNull(careOrder.getCourtName(), defaultCourtName),
+                careOrder.getDateOfIssue());
```

Comment:
Could we have separate methods to generate text? I know it's not big improvement but will be easier when FPL will need to suport Welsh.. Maybe dedicated service?

Reply:
We can think about future features once requirements are known